src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -5437,6 +5462,1262 @@ class CL1<T>
                 );
         }
 
+        [Fact]
+        public void MakeMethodKeyForWhereMethod()
+        {
+            // this test verifies that a bad method symbol doesn't crash when generating a key for external annotations
+            CSharpCompilation c = CreateCompilation(@"
+class Test
+{
+    public void SimpleWhere()
+    {
+        int[] numbers = { 1, 2, 3 };
+        var lowNums = from n in numbers
+                      where n < 5
+                      select n;
+    }
+}", parseOptions: TestOptions.Regular8);
+
+            c.VerifyDiagnostics(
+                // (7,33): error CS1935: Could not find an implementation of the query pattern for source type 'int[]'.  'Where' not found.  Are you missing a reference to 'System.Core.dll' or a using directive for 'System.Linq'?
+                //         var lowNums = from n in numbers
+                Diagnostic(ErrorCode.ERR_QueryNoProviderStandard, "numbers").WithArguments("int[]", "Where").WithLocation(7, 33)
+                );
+        }
+
+        [Fact]
+        public void NotNullWhenFalse_Simple()
+        {
+            CSharpCompilation c = CreateCompilation(@"
+using System.Runtime.CompilerServices;
+public class C
+{
+    public void Main(string? s)
+    {
+        if (MyIsNullOrEmpty(s))
```

Comment:
>MyIsNullOrEmpty(s) [](start = 12, length = 18)

Please add a test for conditions of the form ```MyIsNullOrEmpty(s) == true``` and ```MyIsNullOrEmpty(s) == false```. And also with comparison operands in different order.

Reply:
That is indeed broken. Thanks!
Add tests `NotNullWhenFalse_ComparedToTrue` and `NotNullWhenFalse_ComparedToFalse`

---
In reply to: [188023547](https://github.com/dotnet/roslyn/pull/26656#discussion_r188023547) [](ancestors = 188023547)